freedomofpress/securethenews
# onion service scanning: check meta tags for onion-location

Closes #269. Followup to #262.

This PR:
- Changes onion-location to a nullable field. This is motivated primarily by wanting to have a way to distinguish between failed and successful scans, both for anyone who were to consume this data but also for the STN Twitter bot (that code is [here](https://github.com/thisisparker/stn-tweets)). 
- Gets the page content using `requests`, parses the page content using `lxml`, then sees if there is an `onion-location` meta tag

The easiest way to test once on this branch is to:
1. Ensure you have the test data added (so you have some sites to test on). Then run a scan: `make dev-scan`
2. Visit the API: `http://127.0.0.1:8000/api/v1/sites/`
3. For any sites that failed during the scan, they should have a null `onion_available`. Otherwise, Propublica should have `onion_available` as true, and techcrunch should have `onion_available` as false.
